weather/contrib/ecowitt/server.py

```diff
@@ -275,11 +289,19 @@ def fetchWeather():
                     if INFLUX:
                         log.debug("Writing to InfluxDB")
                         try:
-                            client = InfluxDBClient(host=IHOST,
-                                port=IPORT,
-                                username=IUSER,
-                                password=IPASS,
-                                database=IDB)
+                            if ITOKEN == "":
+                                # Influx 1.8
+                                client = InfluxDBClient(host=IHOST,
+                                    port=IPORT,
+                                    username=IUSER,
+                                    password=IPASS,
+                                    database=IDB)
+                            else:
+                                # Influx 2.x
+                                client = InfluxDBClient(
+                                    url=IURL,
+                                    token=ITOKEN,
+                                    org=IORG)
                             output = [{}]
                             output[0]["measurement"] = IFIELD
                             output[0]["time"] = int(currentts)
```

Comment:
I had to change this line - I don't know if this is an issue with the new client or just best practice:

```python
# for weathe411
# was: output[0]["time"] = weather["dt"]
# now:
output[0]["time"] = datetime.utcfromtimestamp(weather["dt"])

# yours?
output[0]["time"] = datetime.utcfromtimestamp(int(currentts))
```

Reply:
I had a look at that change in Weather411, but I had a difference between ecowitt and weather411 in this section in the original version - Weather411 has a date that can be parsed, whereas ecowitt doesn't, which is why my version just passes in a truncated (int) timestamp, so I didn't think it needed to be changed.

I'll have a look for a missing from datetime import datetime, but I would have thought that would have shown up in a diff - perhaps I wasn't paying attention :)

But the bigger issue is the missing api_client.

Anyway something for another day.